enhancements/ingress/aws-load-balancer-operator.md

```diff
@@ -0,0 +1,394 @@
+    AWS’s newer services.
+2. **Network Load Balancer:** This is also a Level 4 load balancer but has been
+    optimized for performance and cost compared to the Classic Load Balancer.
+3. **Application Load Balancer:** This is a Level 7 load balancer which can
+    route requests based on the HTTP path and host. It also integrates with other
+    AWS services like DDoS protection and WAF.
+
+### Openshift Router
+In OCP/OKD Ingress resources are serviced by the Openshift
+[router](https://github.com/openshift/router). Routers are created by the
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator/)(CIO)
+for every
+_[IngressController](https://docs.openshift.com/container-platform/4.9/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-ingress-controller.html)_
+resource. By default on AWS the CIO creates a Service of type
+[LoadBalancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer).
+The in-tree Service controller in turn provisions a Network Load
+Balancer with a TCP listener. The CIO then creates a DNS entry for the subdomain
+specified in the IngressController resource with the load balancer created in
+the previous step as the target. 
+
+### aws-load-balancer-controller
+AWS maintains an out-of-tree controller with support for ALBs for Ingress and
+Service type resources. The in-tree Service controller only provisions [classic
+load
+balancers](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/introduction.html)
+and
+[Network](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html)
+Load Balancers. The out-of-tree controller supports integration of Kubernetes
+resources with more of the AWS services like WAFs and the DDoS prevent service
+[Shield](https://docs.aws.amazon.com/waf/latest/developerguide/ddos-overview.html).
+It also enables traffic routing from the load balancer directly to the pods of
+an Ingress or Service if the cluster uses the [AWS VPC Container Networking
+Interface](https://github.com/aws/amazon-vpc-cni-k8s). In addition it also
+introduces 2 Custom Resources namely
+[`IngressClassParams`](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.3/guide/ingress/ingress_class/#ingressclassparams)
+and
+[`TargetGroupBindings`](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.3/guide/targetgroupbinding/targetgroupbinding/).
+_IngressClassParams_ is used for specifying custom parameters for Ingresses of a
+certain class.  The controller uses _TargetGroupBinding_ internally and it can
+also be used by users to manage the load balancer externally. The
+aws-load-balancer will henceforth be referred to as **“lb-controller”** for the
+sake of brevity.
+
+## Motivation
+
+Implement an operator, tentatively called `aws-load-balancer-operator`, which
+deploys and manages an instance of the lb-operator. The operator will be
```

Comment:
The "lb-operator" in this paragraph is actually the "lb-controller" defined in the previous paragraph, correct?  
```suggestion
deploys and manages an instance of lb-controller. The operator will be
```